Content.Server/Voting/VoteCommands.cs

```diff
@@ -129,6 +191,88 @@ public CompletionResult GetCompletion(IConsoleShell shell, string[] args)
             var n = args.Length - 1;
             return CompletionResult.FromHint(Loc.GetString("cmd-customvote-arg-option-n", ("n", n)));
         }
+
+        // Sends the payload's message. 
+        private async void WebhookMessage(WebhookPayload payload, string id)
+        {
+            if(String.IsNullOrEmpty(_webhookUrl))
+                return;
+
+            if(id == string.Empty)
+            {
+                var request = await _httpClient.PostAsync($"{_webhookUrl}?wait=true",
+                    new StringContent(JsonSerializer.Serialize(payload), Encoding.UTF8, "application/json"));
+
+                var content = await request.Content.ReadAsStringAsync();
+                webhookId = (string) JsonNode.Parse(content)?["id"]!;
+            } else {
+                await _httpClient.PatchAsync($"{_webhookUrl}/messages/{id}",
+                    new StringContent(JsonSerializer.Serialize(payload), Encoding.UTF8, "application/json"));
+            }
+        }
+
+        public struct WebhookPayload
+        {
+            [JsonPropertyName("username")]
+            public string Username { get; set; } = "";
+
+            [JsonPropertyName("avatar_url")]
+            public string AvatarUrl { get; set; } = "";
+
+            [JsonPropertyName("embeds")]
+            public List<Embed>? Embeds { get; set; } = null;
+
+            public WebhookPayload() { }
+        }
```

Comment:
arent these duplicated from the ahelp webhook stuff? can it not just share the same structs

Reply:
No, since this also uses Fields and just generally includes different info to get it to look the way it does

Reply:
See https://github.com/space-wizards/space-station-14/pull/19468 for a more generic webhook api once that is merged